doc/source/data/loading-data.rst

```diff
@@ -928,16 +958,19 @@ Loading other datasources
 
 If Ray Data can't load your data, subclass
 :class:`~ray.data.Datasource`. Then, construct an instance of your custom
-datasource and pass it to :func:`~ray.data.read_datasource`.
+datasource and pass it to :func:`~ray.data.read_datasource`. To write results, you might
+also need to subclass :class:`ray.data.Datasink`. Then, create an instance of your custom
+datasink and pass it to :func:`~ray.data.Dataset.write_datasink`. For more details, see the guide
+:ref:`Advanced: Read and Write Custom File Types <custom_datasource>`.
 
 .. testcode::
     :skipif: True
 
     # Read from a custom datasource.
     ds = ray.data.read_datasource(YourCustomDatasource(), **read_args)
 
-    # Write to a custom datasource.
-    ds.write_datasource(YourCustomDatasource(), **write_args)
+    # Write to a custom datasink.
+    ds.write_datasink(YourCustomDatasink())
 
 Performance considerations
 ==========================
```

Comment:
Could you also incorporate the [Optimizing read](https://anyscale-ray--44093.com.readthedocs.build/en/44093/data/performance-tips.html#optimizing-reads) section to here. 

Also, we need to add a tip on how to deal with failure retries.